src/core.jl

```diff
@@ -2,7 +2,7 @@
 # Reference:
 # Blinn, J. (1996). Consider the lowly 2 x 2 matrix. IEEE Computer Graphics and Applications, 16(2), 82-88.
 # https://scicomp.stackexchange.com/questions/8899/robust-algorithm-for-2-times-2-svd
-function svd2x2(M::AbstractArray)
+function svd2x2(M::AbstractMatrix{T}) where T <: Number
```

Comment:
Farneback algorithm uses matrix views with this function.
I guess it can be changed in a separate PR if needed, to keep this PR from growing in size even more.

Reply:
Never mind, I just realize that it actually makes no differences at all:

```julia
julia> @btime svd2x2($(SMatrix{2,2}(rand(2, 2))));
  43.171 ns (0 allocations: 0 bytes)

julia> @btime svd2x2($(rand(2, 2)));
  43.698 ns (0 allocations: 0 bytes)
```